### ISSUE TITLE
[Documentation] Docker-CE installation on RHEL

### ISSUE THREAD

**Issue by NBardelot** [docker/docker.github.io#3818]
As far as I know : 
  * Docker-CE is supported on CentOS
  * Docker-EE is supported on RHEL and CentOS
  * Docker-CE is *not* supported on RHEL

But in reality nothing prevents the installation of Docker-CE under RHEL. Thus, adding a little caveat in the installation documentation of Docker-CE would really make our life easier, so that we can convince our CTOs that they should take Docker seriously.

An example of what I would expect in the Docker-CE installation guide would be: 

> Please notice that while Docker-CE is not supported for RHEL, if you still want to install it the RedHat 'extras' channel is as of version 17.06 a prerequisite that provides the *container-selinux* package.

**Comment by anthonyliriano** [comment]
I agree! I provide consulting services to several IT teams and Docker has been a difficult conversation due this roadblock. 

People always assume that they CAN'T install the software in RHEL environment unless they pay for Docker CE.

**Comment by marcellodesales** [comment]
Can you please clearly add instructions how to add the `extras` channel??? I'm on a fire right now and I can't find it right away...

**Comment by mstanleyjones** [comment]
Docker CE is not supported on RHEL, full-stop. We do not document unsupported configurations.

cc/ @friism

**Comment by marcellodesales** [comment]
True, forgot about that point...

**Comment by NBardelot** [comment]
@mstanleyjones as stated in the topic, I know Docker-CE is not officially supported on RHEL, which is absolutely weird but I'm not here to discuss that point. Whatsoever, putting a caveat in the documentation would neither be "supporting" nor "documenting" (at least not in the sense of a "here's the installation documentation"). It would only warn people so that they benefit from other people experience.

In other words, don't look at my proposition from a technical point of view, but from a user experience  / community point-of-view: don't make everyone browse StackOverflow to find the answer to a known asked question.

**Comment by friism** [comment]
@NBardelot thanks for caring about this.

I'm not sure that Docker is going to be taken more seriously by CTOs if we document (or hint at) how to install a package on an operating system that the package wasn't built for nor tested on and where the package is not supported.

If you want to install Docker on RHEL (which has a free trial), please follow the docs here: https://docs.docker.com/engine/installation/linux/docker-ee/rhel/#prerequisites

**Comment by hatdropper1977** [comment]
I have this issue on Centos 7 - which IS supported.

```bash
$ uname -sr
Linux 3.10.0-514.21.1.el7.x86_64
$ cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core) 
$ sudo yum-config-manager --enable extras
```

When I try to install docker-ce I get this error:

```bash
Error: Package: docker-ce-17.06.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
           Requires: container-selinux >= 2.9
```

**Comment by andrewhsu** [comment]
@hatdropper1977 the `extras` repo should be enabled by default in a fresh install of centos 7.3.1611. You can verify the repo is enabled by listing the repos:
```
# yum repolist
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
 * base: centos.mbni.med.umich.edu
 * extras: mirror.hostduplex.com
 * updates: centos.sonn.com
repo id                                       repo name                                       status
base/7/x86_64                                 CentOS-7 - Base                                 9363
extras/7/x86_64                               CentOS-7 - Extras                                450
updates/7/x86_64                              CentOS-7 - Updates                              2146
repolist: 11959
```

If it is not enabled, perhaps have a look in the `/etc/yum.repos.d/CentOS-Base.repo` file to see if the `[extras]` repo definition section had a `enabled=0` attribute added that turns it off.

**Comment by andrewhsu** [comment]
@hatdropper1977 to follow on my previous comment, if `extras` repo is enabled, the `container-selinux` can be resolved when installing docker-ce.

**Comment by hatdropper1977** [comment]
Yes, that was the issue.  Not related to docker but rather the
configuration of my local extras repo

**Comment by friism** [comment]
@hatdropper1977 thx - was the misconfiguration what @andrewhsu suggested? Maybe we can check for that somehow, in the install flow @mstanleyjones

**Comment by mstanleyjones** [comment]
This is a tough one. `extras` is enabled by default on CentOS and we do document that it needs to be enabled. If the user has done something creative with their OS, this is an unbounded thing for us to try to solve. Prerequisites are listed so you can be sure you meet them.

**Comment by hatdropper1977** [comment]
@friism The SA who developed the local repo omitted **_extras_**.  Probably not an every day occurance.

**Comment by friism** [comment]
alright, thanks!

**Comment by bbaassssiiee** [comment]
After installing docker-ce on RHEL expect yourself in a rabbit hole of python library conflicts of stuff installed by yum and pip (docker-compose, docker libraries and dependencies).

**Comment by almoskvin** [comment]
For those, who are eager to install Docker on RHEL, please, consider the official Red Hat documentation [https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html-single/getting_started_with_containers/index#getting_docker_in_rhel_7](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html-single/getting_started_with_containers/index#getting_docker_in_rhel_7)
